Feat(eos_cli_config_gen): Global IP Locking Configuration Options

emilarista opened the pull request:
## Change Summary

Adds data model for global IP locking configuration options.

## Related Issue(s)

Fixes #2313 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
```yaml
address_locking:
  dhcp_servers_ipv4:
    - x.x.x.x
  disabled: bool
  leases:
    - ip: y.y.y.y
      mac: hhhh.hhhh.hhhh
    - ip: z.z.z.z
      mac: hh:hh:hh:hh:hh:hh
  local_interface:  str
  locked_address:
    expiration_mac_disabled: bool
    ipv4_enforcement_disabled: bool
    ipv6_enforcement_disabled: bool
```

## How to test
Tested with molecule and on physical EOS device

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

emilarista committed:
working on address_locking

emilarista committed:
Doc template and adjustments

emilarista committed:
dummy change to retrigger CI

emilarista committed:
fixed small issue in doc template

onurgashi reviewed (approved):
LGTM!

gmuloc commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/address-locking.j2:
```diff
@@ -0,0 +1,49 @@
+
+## Address Locking Summary
+
+| Settings | Value |
+| -------- | ----- |
+| Disable IP locking on configured ports | {{ address_locking.disabled | arista.avd.default(false) }} |
+{%     if address_locking.local_interface is arista.avd.defined %}
+| Local Interface | {{ address_locking.local_interface | arista.avd.default("-") }} |
+{%     endif %}
+{%     if address_locking.locked_address.expiration_mac_disabled is arista.avd.defined(true) %}
+| Disable deauthorizing locked addresses upon MAC aging out | True |
+{%     endif %}
+{%     if address_locking.locked_address.ipv4_enforcement_disabled is arista.avd.defined(true) %}
+| Disable enforcement for locked ipv4 addresses | True |
+{%     endif %}
+{%     if address_locking.locked_address.ipv6_enforcement_disabled is arista.avd.defined(true) %}
+| Disable enforcement for locked ipv6 addresses | True |
+{%     endif %}
+{%     if address_locking.dhcp_servers_ipv4 is arista.avd.defined %}
```
Maybe let's make these bullet points rather than a table like we do in other templates?

if not then at `Settings` -> `Setting` for the header

ClausHolbechArista replied:
It varies a lot between the different templates. We have a mix of both.

gmuloc replied:
ok so whichever is more appropriate here - in any case let's fix the header if we keep it like this

gmuloc reviewed (commented):
Overall looks good to me

one comment about the documentation where I feel we could use bullet points instead of the table of settings.

Also just for reference / posterity - this PR covers everything available today in EOS:

```
Leaf1(config-address-locking)#show cli commands
APPEND [ COMMENT ]
[no|default] comment
[no|default] dhcp server ipv4 V4ADDR
[no|default] disabled
do CMD
exit
[no|default] lease ( V4ADDR | V6ADDR ) mac MAC_ADDR
[no|default] local-interface INTF
[no|default] locked-address ( ipv4 | ipv6 ) enforcement disabled
[no|default] locked-address expiration mac disabled
show active
show active all [ detail ]
show comment
```

ClausHolbechArista committed:
Update address-locking.j2

- Update table header to singular.
- Document both true and false values when set.
- Remove unneeded default values.
- Don't print disable ip locking setting when not set.

ClausHolbechArista committed:
Minor update and rerun molecule

ClausHolbechArista reviewed (approved):
I have updated with the new documentation headings - We have added one `#` on all headings.
